File: src/main/java/gg/mineral/practice/inventory/menus/QueuetypeArenaEnableMenu.java
package gg.mineral.practice.inventory.menus;

import org.bukkit.ChatColor;
import org.bukkit.inventory.ItemStack;

import gg.mineral.practice.arena.Arena;
import gg.mineral.practice.inventory.ClickCancelled;
import gg.mineral.practice.inventory.PracticeMenu;
import gg.mineral.practice.managers.ArenaManager;
import gg.mineral.practice.queue.Queuetype;
import gg.mineral.practice.util.items.ItemBuilder;
import gg.mineral.practice.util.messages.CC;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import it.unimi.dsi.fastutil.objects.ObjectCollection;
import lombok.RequiredArgsConstructor;

@ClickCancelled(true)
@RequiredArgsConstructor
public class QueuetypeArenaEnableMenu extends PracticeMenu {
    private final Queuetype queuetype;

    @Override
    public void update() {
        clear();

        ObjectCollection<Arena> arenas = ArenaManager.getArenas().values();

        for (Arena a : arenas) {
            boolean arenaEnabled = queuetype.getArenas().contains(a.getId());
            ChatColor color = arenaEnabled ? ChatColor.GREEN : ChatColor.RED;

            ItemStack item;
            try {
                item = new ItemBuilder(a.getDisplayItem())
                        .name(CC.SECONDARY + CC.B + a.getDisplayName()).lore(color.toString() + arenaEnabled).build();
            } catch (Exception e) {
                continue;
            }

            add(item, interaction -> {
                queuetype.enableArena(a, !arenaEnabled);
                ChatMessages.QUEUETYPE_ARENA_SET.clone().replace("%queuetype%", queuetype.getName())
                        .replace("%toggled%", "" + !arenaEnabled).replace("%arena%", a.getName())
                        .send(viewer.getPlayer());
                reload();
            });
        }
    }

    @Override
    public String getTitle() {
        return CC.BLUE + "Toggle Arena";
    }

    @Override
    public boolean shouldUpdate() {
        return true;
    }
}

File: src/main/kotlin/gg/mineral/practice/inventory/menus/QueuetypeArenaEnableMenu.kt
package gg.mineral.practice.inventory.menus

import gg.mineral.practice.inventory.ClickCancelled
import gg.mineral.practice.inventory.PracticeMenu
import gg.mineral.practice.managers.ArenaManager.arenas
import gg.mineral.practice.queue.Queuetype
import gg.mineral.practice.util.items.ItemBuilder
import gg.mineral.practice.util.messages.CC
import gg.mineral.practice.util.messages.impl.ChatMessages
import org.bukkit.ChatColor
import org.bukkit.inventory.ItemStack

@ClickCancelled(true)
class QueuetypeArenaEnableMenu(val queuetype: Queuetype) : PracticeMenu() {

    override fun update() {
        clear()

        val arenas = arenas.values

        for (a in arenas) {
            a ?: continue
            val arenaEnabled = queuetype.arenas.contains(a.id)
            val color = if (arenaEnabled) ChatColor.GREEN else ChatColor.RED

            val item: ItemStack
            try {
                item = ItemBuilder(a.displayItem)
                    .name(CC.SECONDARY + CC.B + a.displayName).lore(color.toString() + arenaEnabled).build()
            } catch (e: Exception) {
                continue
            }

            add(item) {
                queuetype.enableArena(a, !arenaEnabled)
                viewer.message(
                    ChatMessages.QUEUETYPE_ARENA_SET.clone().replace("%queuetype%", queuetype.name)
                        .replace("%toggled%", "" + !arenaEnabled).replace("%arena%", a.name)
                )
                reload()
            }
        }
    }

    override val title: String
        get() = CC.BLUE + "Toggle Arena"

    override fun shouldUpdate() = true
}
